libs/execution_context/ebpf_core.c

```diff
@@ -1207,10 +1207,19 @@ ebpf_core_get_pinned_object(_In_ const cxplat_utf8_string_t* path, _Out_ ebpf_ha
     }
 
     retval = ebpf_handle_create(handle, (ebpf_base_object_t*)object);
+    if (retval != EBPF_SUCCESS) {
+        EBPF_LOG_MESSAGE_ERROR(
+            EBPF_TRACELOG_LEVEL_ERROR, EBPF_TRACELOG_KEYWORD_CORE, "ebpf_core_get_pinned_object failed", retval);
+    }
 
 Done:
     EBPF_OBJECT_RELEASE_REFERENCE((ebpf_core_object_t*)object);
-    EBPF_RETURN_RESULT(retval);
+    // Upon error, callers of this function will log the error.
+    if (retval == EBPF_SUCCESS) {
+        EBPF_LOG_FUNCTION_SUCCESS();
+    }
+    EBPF_LOG_EXIT();
```

Comment:
Could be collapsed into one line?

```suggestion
    EBPF_RETURN_RESULT(retval);
```